pkg/providers/instancetype/instancetype.go

```diff
@@ -105,6 +107,13 @@ func (p *Provider) List(ctx context.Context, kc *corev1beta1.KubeletConfiguratio
 		return aws.StringValue(s.AvailabilityZone)
 	})...)
 
+	if kc == nil {
```

Comment:
If we make this change up here, do we still need the `nil` checks down below for whether kubeletConfig is set or not? (This probably becomes less of an issue when you rebase this PR with some of your other changes)